src/test/mappingHandlers.test.ts

```diff
@@ -1,29 +1,306 @@
+    ),
+  ],
+
+  'handleStateChangeEvent',
+);
+
+const vault = new Vault(
+  'published.vaultFactory.managers.manager0.vaults.vault6',
+  BigInt(742),
+  new Date('2024-12-09 23:13:19.186'),
+  'published.wallet.agoric1ee9hr0jyrxhy999y755mp862ljgycmwyp4pl7q.current',
+);
+
+vault.balance = BigInt(788000000);
+vault.lockedValue = BigInt(788000000);
+vault.coin = 'ATOM';
+vault.denom = 'ATOM';
+vault.debt = BigInt(4745610000);
+vault.state = 'active';
+
+subqlTest(
+  'Vault and VaultStateDailies at height 742',
+  742,
+  [],
+  [
+    vault,
+    new VaultStatesDaily(
+      '20241209',
+      BigInt(742),
+      new Date('2024-12-09 23:13:19.186'),
+      BigInt(1),
+      BigInt(0),
+      BigInt(0),
+      BigInt(0),
+      BigInt(0),
+    ),
+  ],
+
+  'handleStateChangeEvent',
+);
+
+const reserveAllocationMetricsDaily = new ReserveAllocationMetricsDaily(
+  'IST:20241216',
+  'IST',
+  20241216,
+  BigInt(1212),
+  new Date('2024-12-16T11:42:27.000Z'),
+);
+
+reserveAllocationMetricsDaily.denom = 'IST';
+reserveAllocationMetricsDaily.key = 'Fee';
+reserveAllocationMetricsDaily.valueLast = BigInt(184980868);
+reserveAllocationMetricsDaily.metricsCount = BigInt(1);
+
+const vaultManagerMetricsDaily = new VaultManagerMetricsDaily(
+  'published.vaultFactory.managers.manager1.metrics:20241216',
+  'published.vaultFactory.managers.manager1.metrics',
+  20241216,
+  BigInt(1212),
+  new Date('2024-12-16 11:42:27.202'),
+);
+
+vaultManagerMetricsDaily.liquidatingCollateralBrand = 'stATOM';
+vaultManagerMetricsDaily.liquidatingDebtBrand = 'IST';
+vaultManagerMetricsDaily.liquidatingCollateralValueLast = BigInt(0);
+vaultManagerMetricsDaily.liquidatingDebtValueLast = BigInt(0);
+vaultManagerMetricsDaily.lockedQuoteDenominatorLast = BigInt(0);
+vaultManagerMetricsDaily.lockedQuoteNumeratorLast = BigInt(0);
+vaultManagerMetricsDaily.numActiveVaultsLast = BigInt(0);
+vaultManagerMetricsDaily.numLiquidatingVaultsLast = BigInt(0);
+vaultManagerMetricsDaily.numLiquidationsAbortedLast = BigInt(0);
+vaultManagerMetricsDaily.numLiquidationsCompletedLast = BigInt(0);
+vaultManagerMetricsDaily.retainedCollateralLast = BigInt(0);
+vaultManagerMetricsDaily.totalCollateralLast = BigInt(0);
+vaultManagerMetricsDaily.totalCollateralSoldLast = BigInt(0);
+vaultManagerMetricsDaily.totalDebtLast = BigInt(0);
+vaultManagerMetricsDaily.totalOverageReceivedLast = BigInt(0);
+vaultManagerMetricsDaily.totalProceedsReceivedLast = BigInt(0);
+vaultManagerMetricsDaily.totalShortfallReceivedLast = BigInt(0);
+vaultManagerMetricsDaily.metricsCount = BigInt(1);
+
+// DISABLED because the timestamp for these entities is changed on each run of the a3p container
```

Comment:
are you saying the time for the block (e.g. 1212) changes with each run?

any ideas why these paths have that problem and the ones below like boardAux don't?


I looked into expressing the tests to not expect a certain date but the subqlTest runner expects a full exact match: https://github.com/subquery/subql/blob/cb2db57fa3be7a3e4033cc170ea5d8f45fadf90c/packages/node-core/src/indexer/test.runner.ts#L96-L107

thankfully we can instead automate testing using GraphQL queries as you've done in  https://github.com/Agoric/agoric-subql/pull/36

Reply:
yes. it seems the blocks that contain the data for vault metrics are committed _after_ the chain is started, therefore the time the block is committed is different on each time the a3p container is started.

#36 seems like a good approach for this. i'll see what i can do 👍